OpenSSL.Jenkinsfile

```diff
@@ -0,0 +1,84 @@
+pipeline {
+    environment {
+        OPENSSL_VERSION="1.1.1"
+        OPENSSL_BRANCH='OpenSSL_1_1_1-stable'
+        CONFIGURE_PARAMS="no-idea no-camellia no-seed no-bf no-cast no-des no-rc2 no-rc4 no-rc5 \
+        no-md2 no-md4 no-ripemd no-mdc2 no-dsa no-dh no-ec no-ecdsa no-ecdh no-sock \
+        no-ssl2 no-ssl3 no-err no-engine no-hw"
+    }
+    agent {
+        label 'master'
+    }
+    stages {
+        stage('build') {
+            parallel {
+                stage('build linux') {
+                    agent {
+                        label 'linux'
+                    }
+                    steps {
+                   		buildOpenssl(platform: "linux", config: "")
+                    }
+                }
+                stage('build mac') {
+                    agent {
+                        label 'mac'
+                    }
+                    steps {
+						buildOpenssl(platform: "mac", config: "")
+                    }
+                }
+                stage('cross compile for windows') {
+                    agent {
+                        label 'linux'
+                    }
+                    steps {
+                    	echo "ignoring windows for now"
+//                     	buildOpenssl(platform: "win", config: "--cross-compile-prefix=x86_64-w64-mingw32- mingw64")
+                    }
+                }
+            }
+        }
+        stage('publish to nexus') {
+            steps {
+				script {
+					publish("linux")
+					publish("mac")
+// 					publish("win")
+				}
+            }
+        }
+    }
+}
+
+def buildOpenssl(Map params) {
+	sh "rm -rf openssl" // Jenkins doesn't seem to use a clean workspace each time
```

Comment:
you can set a flag to force wipe each time

Reply:
do we use it somewhere else?

Reply:
Yes, we do it at least for admin client, I would recommend always using that